frame/staking/src/lib.rs

```diff
@@ -759,6 +759,43 @@ impl<T: Trait> SessionInterface<<T as frame_system::Trait>::AccountId> for T whe
 	}
 }
 
+pub mod weight {
+	use super::*;
+
+	/// All weight notes are pertaining to the case of a better solution, in which we execute
+	/// the longest code path.
+	/// Weight: 0 + (35 μs * v) + (25 μs * n)
+	/// State reads:
+	/// 	- Initial checks:
+	/// 		- ElectionState, CurrentEra, QueuedScore
+	/// 		- SnapshotValidators.len()
+	/// 		- ValidatorCount
+	/// 		- SnapshotValidators
+	/// 		- SnapshotNominators
+	/// 	- Iterate over nominators:
+	/// 		- compact.len() * Nominators(who)
+	/// 		- (non_self_vote_edges) * SlashingSpans
+	/// 	- For `assignment_ratio_to_staked`: Basically read the staked value of each stash.
+	/// 		- (winners.len() + compact.len()) * (Ledger + Bonded)
+	/// 		- TotalIssuance (read a gzillion times potentially, but well it is cached.)
+	/// - State writes:
+	/// 	- QueuedElected, QueuedScore
+	pub fn weight_for_submit_solution<T: Trait>(
+		winners: &Vec<ValidatorIndex>,
+		compact: &CompactAssignments,
+		size: &ElectionSize,
+	) -> Weight {
+		(35 * WEIGHT_PER_MICROS * (size.validators as Weight))
+			.saturating_add(25 * WEIGHT_PER_MICROS * (size.nominators as Weight))
```

Comment:
```suggestion
		(35 * WEIGHT_PER_MICROS).saturating_mul(size.validators as Weight)
			.saturating_add((25 * WEIGHT_PER_MICROS).saturating_mul(size.nominators as Weight))
```
`size` is a user input, right?